olmo/optim.py

```diff
@@ -480,7 +481,11 @@ def get_max_grad_norm_ratio(
         return self._get_max_grad_norm_coeff(initial_max_grad_norm_ratio, step, max_steps)
 
     def _linear_warmup(self, initial_lr: float, step: int, warmup_steps: int = 2000) -> float:
-        return initial_lr * (0.1 + 0.9 * min(step, warmup_steps) / warmup_steps)
+        if self.warmup_min_lr is not None:
+            assert initial_lr > self.warmup_min_lr
```

Comment:
nit: You could also add a `assert self.warmup_min_lr >= 0` for good measure

Reply:
Done: 75eba56a